trajopt_ifopt/src/cartesian_position_constraint.cpp

```diff
@@ -0,0 +1,120 @@
+#include <trajopt_ifopt/constraints/cartesian_position_constraint.h>
+
+#include <tesseract_kinematics/core/utils.h>
+
+#include <console_bridge/console.h>
+
+namespace trajopt
+{
+CartPosConstraint::CartPosConstraint(const Eigen::Isometry3d& target_pose,
+                                     CartPosKinematicInfo::ConstPtr kinematic_info,
+                                     JointPosition::Ptr position_var,
+                                     const std::string& name)
+  : ifopt::ConstraintSet(6, name)
+  , position_var_(std::move(position_var))
+  , target_pose_(target_pose)
+  , target_pose_inv_(target_pose.inverse())
+  , kinematic_info_(std::move(kinematic_info))
+{
+  // Set the n_dof and n_vars for convenience
+  n_dof_ = kinematic_info_->manip_->numJoints();
+  assert(n_dof_ > 0);
+
+  bounds_ = std::vector<ifopt::Bounds>(6, ifopt::BoundZero);
+}
+
+Eigen::VectorXd CartPosConstraint::GetValues() const
+{
+  Eigen::Isometry3d new_pose;
+  VectorXd joint_vals = this->GetVariables()->GetComponent(position_var_->GetName())->GetValues();
+  kinematic_info_->manip_->calcFwdKin(new_pose, joint_vals, kinematic_info_->kin_link_->link_name);
+
+  new_pose = kinematic_info_->world_to_base_ * new_pose * kinematic_info_->kin_link_->transform * kinematic_info_->tcp_;
+
+  Eigen::Isometry3d pose_err = target_pose_inv_ * new_pose;
+  Eigen::VectorXd err = concat(pose_err.translation(), calcRotationalError(pose_err.rotation()));
+  //    Eigen::VectorXd reduced_err(indices_.size());
+  //    for (int i = 0; i < indices_.size(); ++i)
+  //      reduced_err[i] = err[indices_[i]];
```

Comment:
Should this be commented out?

Reply:
I don't think we can do this in this framework since the constraint is expected to be the same size each time. I'll remove them.